### DISCUSSION TITLE
Help with extension endpoint and accountability

### DISCUSSION THREAD

**Discussion by nervusdm** [directus/directus#14850] [Archive]
Using an url like /audit (no /admin/audit?) , even If I am connected, accountability tell me I am not.

What should I do to use the current accountability  ?

```
{
  user: null,
  role: null,
  admin: false,
  app: false,

```

`export default (router, { services, exceptions }) => {
	const { ItemsService } = services;
	const { ServiceUnavailableException } = exceptions;

	router.get('/:id', (req, res, next) => {

console.log(req.accountability, req.accountability.id);
		const recipeService = new ItemsService('hs_audit', { schema: req.schema, accountability: req.accountability });

		recipeService
			.readByQuery({ sort: ['name'], fields: ['*'] })
			.then((results) => res.json(results))
			.catch((error) => {
				return next(new ServiceUnavailableException(error.message));
			});

return ;`

**Comment by br41nslug** [comment]
When using an API endpoint (or custom endpoint) you'll need to provide an access token with the request https://docs.directus.io/reference/authentication/#access-tokens

**Reply by nervusdm** [reply]
Thanks for this answer. I dont know why I have to add the token in the url params (since the user is connected), seems a bit ugly, but I will do this.
Can I send  this params in a button field type ? 
If I use {$token}. that's not working.
![Capture d’écran 2022-08-05 à 10 24 03](https://user-images.githubusercontent.com/4217180/183035372-1cf929dd-0d93-4334-881d-d0c1cee9ae5f.png)
